core/server/src/main/java/alluxio/master/block/BlockMaster.java

```diff
@@ -610,8 +609,12 @@ private void processWorkerRemovedBlocks(MasterWorkerInfo workerInfo,
     for (long removedBlockId : removedBlockIds) {
       MasterBlockInfo masterBlockInfo = mBlocks.get(removedBlockId);
       if (masterBlockInfo == null) {
-        LOG.warn("Worker {} removed block {} but block does not exist.", workerInfo.getId(),
-            removedBlockId);
+        LOG.warn("Worker {} informs the removed block {}, but block metadata does not exist"
+            + "on Master!", workerInfo.getId(), removedBlockId);
+        // TODO(pfxuan): [ALLUXIO-1804] should find a better way to handle the removed blocks.
```

Comment:
If MasterBlockInfo does not exist but Worker has its block, the system will lose the consistency on that block, which should never happen. I'm thinking it's better to refactor current I/O flows (e.g. create, remove, rename and so on) using Raft(Paxos) protocol:  https://alluxio.atlassian.net/browse/ALLUXIO-584